test/integration/good/shadow.stan

```diff
@@ -0,0 +1,7 @@
+parameters {
+  real pow;
+}
+transformed parameters {
+  print(pow(1,1));
+  print(pow);
+}
```

Comment:
Up until not a long time ago stanc3 only allowed shadowing of nullary builtin functions. e(), positive_infinity() and things like that.

That was changed not long ago to allow shadowing for all. Precisely for stuff like beta etc. On mobile right now so cant point to the exact PR.

Will try to find how we could avoid this. Maybe specifiying a namespace would help.

Reply:
Ah, so the comment had just gone stale. I duplicated the logic we currently use but added this test to make sure it would be tested in the future, and it seems like it's a good thing I did so. 

Happy to help fix the issue if you can find out more about it

Reply:
We can make that test better (include something like `pow` and call the functions) and I can remove `shadow.stan` from this PR. LMK

Reply:
Yeah, lets just rename that one to shadow and add pow, add calls to unary, binary functions and even an lpdf.